src/Pirouette/Symbolic/Eval.hs

```diff
@@ -327,19 +352,30 @@ symEvalOneStep t@(R.App hd args) = case hd of
           R.TermArg
           args
 
+-- A-ha... it's the state monad annoying us. Maybe if we can understand whether
+-- we really need the state monad we can better understand the parallelism possibilities
+-- in here. If we do really need the monad, we can only paralellize the prune and pruneAndValidate
+-- functions, if not, we could paralellize everything.
```

Comment:
We are actually using `State` mostly to nest changes to the statistics. In fact, it should be possible to create a version of `ReaderT` which gives you back the "innermost" thing it used for its `local` or something like that.